lib/azure/core/version.rb

```diff
@@ -0,0 +1,33 @@
+#-------------------------------------------------------------------------
+# # Copyright (c) Microsoft and contributors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#--------------------------------------------------------------------------
+
+module Azure
+  module Core
+    class Version
+      MAJOR = 0 unless defined? MAJOR
+      MINOR = 1 unless defined? MINOR
+      UPDATE = 0 unless defined? UPDATE
+      PRE = nil unless defined? PRE
+
+      class << self
+
+        # @return [String]
+        def to_s
+          [MAJOR, MINOR, UPDATE, PRE].compact.join('.')
```

Comment:
Minor: This format doesn't comply with sematic versioning defined in http://semver.org/

quoted: A pre-release version MAY be denoted by appending a **hyphen**